### DISCUSSION TITLE
On Apache

### DISCUSSION THREAD

**Discussion by dvv** [kanidm/kanidm#2503] [Show and tell]
Hi!

I believe I finally managed to make apache behave well )

The lessons (hard) I've learnt:
```apache
    # NB: may be just path, reduces copy-paste
    OIDCRedirectURI /oauth2/callback
    # you must allocate this virtual location matching OIDCRedirectURI and allow it for _any valid user_
    <Location /oauth2/callback>
        AuthType openid-connect
        Require valid-user
    </Location>
    # other locations may be authorized at will
    <Directory /foo>
        AuthType openid-connect
        # see https://github.com/OpenIDC/mod_auth_openidc/wiki/Authorization
        # you can authorize by the groups if you request OIDCScope "openid groups"
        # Require claim groups:<spn | uuid>
        Require claim groups:apache_access_allowed@example.com
        # or authorize by exact preferred_username
        # Require user john.doe
    </Directory>
```

Please consider validating and updating the book if feasible.

TIA

**Comment by Firstyear** [comment]
Feel free to also submit a PR to the book! The book is easy to edit (The source to that section is here https://github.com/kanidm/kanidm/blob/master/book/src/integrations/oauth2.md#apache-mod_auth_openidc ) :) 

It's probably worth showing the `Require claim groups:<spn | uuid>` as a comment above the real line, and changing it from "roots" to something like "apache_access_allowed" or something more descriptive for the example.

**Reply by dvv** [reply]
Made #2524

**Reply by Firstyear** [reply]
Thank you!
